src/java.base/share/classes/java/lang/invoke/BootstrapMethodInvoker.java

```diff
@@ -243,65 +254,54 @@ private static Object invokeWithManyArguments(MethodHandle bootstrapMethod, Look
 
+    /**
+     * Exact type used by the {@link java.lang.invoke.LambdaMetafactory#altMetafactory(
+     * MethodHandles.Lookup,String,MethodType,Object[])} bootstrap method.
+     */
     private static final MethodType LMF_ALT_MT = MethodType.methodType(CallSite.class,
             Lookup.class, String.class, MethodType.class, Object[].class);
 
-    private static final MethodType OBJECT_METHODS_MT = MethodType.methodType(Object.class,
-            Lookup.class, String.class, TypeDescriptor.class, Class.class, String.class, MethodHandle[].class);
-
-    private static final MethodType LMF_CONDY_MT = MethodType.methodType(Object.class,
-            Lookup.class, String.class, Class.class, MethodType.class, MethodHandle.class, MethodType.class);
-
-    private static final MethodType SCF_MT = MethodType.methodType(CallSite.class,
-            Lookup.class, String.class, MethodType.class, String.class, Object[].class);
-
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.StringConcatFactory#makeConcatWithConstants(MethodHandles.Lookup,
-     *                 String,MethodType,String,Object...))}
+     * Exact type used for the {@link java.lang.runtime.ObjectMethods#bootstrap(
+     * MethodHandles.Lookup,String,TypeDescriptor,Class,String,MethodHandle[])} bootstrap method.
      */
-    private static boolean isStringConcatFactoryBSM(MethodType bsmType) {
-        return bsmType == SCF_MT;
-    }
+    private static final MethodType OBJECT_METHODS_MT = MethodType.methodType(Object.class,
+            Lookup.class, String.class, TypeDescriptor.class, Class.class, String.class, MethodHandle[].class);
 
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.LambdaMetafactory#metafactory(
-     *          MethodHandles.Lookup,String,Class,MethodType,MethodHandle,MethodType)}
+     * Exact type of the bootstrap methods generated for {@link java.lang.reflect.Proxy} classes.
      */
-    private static boolean isLambdaMetafactoryCondyBSM(MethodType bsmType) {
-        return bsmType == LMF_CONDY_MT;
-    }
+    private static final MethodType PROXY_GET_METHOD_MT = MethodType.methodType(Method.class,
+            Lookup.class, String.class, Class.class, Class.class, String.class, MethodType.class);
 
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.LambdaMetafactory#metafactory(
-     *          MethodHandles.Lookup,String,MethodType,MethodType,MethodHandle,MethodType)}
+     * Exact type used of the {@link java.lang.invoke.ConstantBootstraps#getStaticFinal(Lookup, String, Class)}
+     * bootstrap method.
      */
-    private static boolean isLambdaMetafactoryIndyBSM(MethodType bsmType) {
-        return bsmType == LMF_INDY_MT;
-    }
+    private static final MethodType CONDY_GET_STATIC_FINAL_MT = MethodType.methodType(Object.class,
```

Comment:
This type should just be called `CONDY_NO_ARG` as it's the most common form of condy that doesn't take any arg (`nullConstant` `primitiveClass` etc.)

Reply:
Note that [`ConstantBootstraps​::primitiveClass​(Lookup, String, Class)`] has a static return type of `Class⁠<⁠?⁠>`, so the following is also needed:
```java
/**
 * Exact type used of the {@link ConstantBootstraps#primitiveClass(Lookup, String, Class)}
 * bootstrap method.
 */
private static final MethodType CONDY_CLASS_NO_ARG_MT
	= CONDY_GET_STATIC_FINAL_MT.changeReturnType(Class.class);
```

[`ConstantBootstraps​::primitiveClass​(Lookup, String, Class)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/invoke/ConstantBootstraps.html#primitiveClass(java.lang.invoke.MethodHandles.Lookup,java.lang.String,java.lang.Class)

Reply:
It's not the intent of this PR to exhaustively enumerate all methods in `ConstantBootstraps`, primarily those shown to be bootstrap sensitive in some app. I've so far never seen a use of `primitiveClass` (and I admit being ignorant as to why this even exists as a BSM) so I don't have any reason to believe special-casing it will carry its own weight.

Reply:
For its existence, I think it was the same as null, that they weren't representable by cp entries for bootstrap method args. They are now backed by PrimitiveClassDescImpl, a type of condy, and ClassFile API currently writes loadConstant(primitiveCd) as a condy.

Reply:
I am sure there's a perfectly good explanation why we've done it like that rather than translating things like javac would (AFAICT a load from `Integer/Short/Byte/...TYPE`).